src/charm.py

```diff
@@ -467,6 +475,42 @@ def _has_required_fields(self, conf_or_rel: _ConfigOrRelation):
         field_names = [f'_{f.replace("-", "_")}' for f in REQUIRED_INGRESS_RELATION_FIELDS]
         return all(getattr(conf_or_rel, f) for f in field_names)
 
+    def _delete_unused_services(self, current_svc_names: List[str]):
+        """Delete services and ingresses that are no longer used.
+
+        Args:
+            current_svc_names: service names set by config or relation data
+            current_svc_hostnames: service hostnames set by config or relation data
+        """
+        self.k8s_auth()
+        api = _core_v1_api()
+        created_by_label = f"{CREATED_BY_LABEL}{self.app.name}"
+        all_services = api.list_namespaced_service(
+            namespace=self._namespace, label_selector=created_by_label
+        )
+        all_svc_names = [item.metadata.name for item in all_services.items]
```

Comment:
Do you think would be worth using a generator instead of a list comprehension for a small list? I'm assuming that would be rare to be more than 100 items or even 50 in this case.

Reply:
(I'm asking out of curiosity, I applied your suggestion)

Reply:
If a generator works, why not use it? It is more efficient (has less functionality than a list) and uses less memory